File: src/main/java/boat/torrent/TorrentService.java
package boat.torrent;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;
import java.nio.charset.Charset;
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;
import java.util.concurrent.atomic.AtomicInteger;

public class TorrentService {

    private List<String> trackerList = new ArrayList<>();
    private List<String> releaseTags = new ArrayList<>();

    public List<String> getTrackerUrls() {
        if (trackerList.size() > 0) {
            return trackerList;
        }
        final int maxChars = 1000;
        AtomicInteger currentChars = new AtomicInteger(0);
        List<String> trackerList = new ArrayList<>();
        try {
            InputStream inputStream = TorrentService.class.getResourceAsStream("/trackers.txt");
            InputStreamReader inputStreamReader = new InputStreamReader(inputStream, Charset.defaultCharset());
            BufferedReader bufferedReader = new BufferedReader(inputStreamReader);
            String line;
            while ((line = bufferedReader.readLine()) != null) {
                if (currentChars.get() + line.length() < maxChars) {
                    trackerList.add(line);
                    currentChars.addAndGet(line.length());
                }
            }
            return trackerList;
        } catch (IOException e) {
            return Collections.emptyList();
        }
    }

    public List<String> getAllTrackerUrls() {
        if (trackerList.size() > 0) {
            return trackerList;
        }
        AtomicInteger currentChars = new AtomicInteger(0);
        List<String> trackerList = new ArrayList<>();
        try {
            InputStream inputStream = TorrentService.class.getResourceAsStream("/trackers.txt");
            InputStreamReader inputStreamReader = new InputStreamReader(inputStream, Charset.defaultCharset());
            BufferedReader bufferedReader = new BufferedReader(inputStreamReader);
            String line;
            while ((line = bufferedReader.readLine()) != null) {
                trackerList.add(line);
            }
            return trackerList;
        } catch (IOException e) {
            return Collections.emptyList();
        }
    }

    public List<String> getReleaseTags() {
        if (releaseTags.size() > 0) {
            return releaseTags;
        }
        List<String> releaseTags = new ArrayList<>();
        try {
            InputStream inputStream = TorrentService.class.getResourceAsStream("/releasetags.txt");
            InputStreamReader inputStreamReader = new InputStreamReader(inputStream, Charset.defaultCharset());
            BufferedReader bufferedReader = new BufferedReader(inputStreamReader);
            String line;
            while ((line = bufferedReader.readLine()) != null) {
                    releaseTags.add(line.toLowerCase());
            }
            return releaseTags;
        } catch (IOException e) {
            return Collections.emptyList();
        }
    }

}

File: src/main/java/boat/torrent/TorrentService.kt
package boat.torrent

import boat.utilities.HttpHelper
import java.io.BufferedReader
import java.io.IOException
import java.io.InputStreamReader
import java.nio.charset.Charset
import java.util.concurrent.atomic.AtomicInteger

class TorrentService(val httpHelper: HttpHelper?) {
    private val trackerList: List<String> = ArrayList()
    private val releaseTags: List<String> = ArrayList()
    val trackerUrls: List<String>
        get() {
            if (trackerList.isNotEmpty()) {
                return trackerList
            }
            return try {
                val trackers = httpHelper?.getPage("https://raw.githubusercontent.com/ngosang/trackerslist/master/trackers_all_http.txt")
                trackers?.split("\n")?.filter(String::isNotEmpty) ?: emptyList()
            } catch (e: IOException) {
                emptyList()
            }
        }

    fun getReleaseTags(): List<String> {
        if (releaseTags.isNotEmpty()) {
            return releaseTags
        }
        return try {
            val inputStream = this.javaClass.getResourceAsStream("/releasetags.txt")
            val inputStreamReader = InputStreamReader(inputStream, Charset.defaultCharset())
            inputStreamReader.readText().split("\n").map { it.lowercase() }
        } catch (e: IOException) {
            emptyList()
        }
    }
}